client_plugin/utils/config/config.go

```diff
@@ -70,3 +86,147 @@ func SetDefaults(config *Config) {
+	usingConfigDefaults := false
+	c, err := Load(*configFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			usingConfigDefaults = true // no .conf file, so using defaults
+			c = Config{}
+			setDefaults(&c)
+		} else {
+			panic(fmt.Sprintf("Failed to load config file %s: %v",
+				*configFile, err))
+		}
+	}
+
+	path := c.LogPath
+	if path == "" {
+		path = defaultLogFile
+	}
+
+	if logFile != nil {
+		path = *logFile
+	}
+	log.SetOutput(&lumberjack.Logger{
+		Filename: path,
+		MaxSize:  c.MaxLogSizeMb,  // megabytes
+		MaxAge:   c.MaxLogAgeDays, // days
+	})
+
+	if *logLevel == "" {
+		*logLevel = c.LogLevel
+	}
+
+	level, err := log.ParseLevel(*logLevel)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to parse log level: %v", err))
+	}
+
+	log.SetFormatter(new(log_formatter.VmwareFormatter))
+	log.SetLevel(level)
+
+	if usingConfigDefaults {
+		log.Info("No config file found. Using defaults.")
+	}
+	return usingConfigDefaults
+}
+
+// InitConfig set up driver specific options
+func InitConfig(defaultConfigPath string, defaultLogPath string, defaultDriver string,
+	defaultWindowsDriver string) (Config, error) {
+	// Get options from ENV (where available), and from command line.
+	// ENV takes precedence, so we can modify it in Docker plugin install
+	logEnv := os.Getenv("VDVS_LOG_LEVEL")
+	logLevel := &logEnv
+	if *logLevel == "" {
+		logLevel = flag.String("log_level", "", "Logging Level")
+	}
+	configFile := flag.String("config", defaultConfigPath, "Configuration file path")
+	driverName := flag.String("driver", "", "Volume driver")
+
+	// Photon driver options
+	targetURL := flag.String("target", "", "Photon controller URL")
+	projectID := flag.String("project", "", "Project ID of the docker host")
+	vmID := flag.String("host", "", "ID of docker host")
+
+	// vSphere driver options
+	port := flag.Int("port", defaultPort, "Default port to connect to ESX service")
+	useMockEsx := flag.Bool("mock_esx", false, "Mock the ESX service")
+
+	flag.Parse()
+
+	// Load the configuration if one was provided.
+	c, err := Load(*configFile)
+	if err != nil {
+		log.Warningf("Failed to load config file %s: %v", *configFile, err)
+	}
+
+	LogInit(logLevel, nil, defaultLogPath, configFile)
+
+	// If no driver provided on the command line, use the one in the
+	// config file or the default.
+	if *driverName != "" {
+		c.Driver = *driverName
+	} else if c.Driver == "" {
+		c.Driver = defaultDriver
+	}
+
+	// The windows plugin only supports the vsphere driver.
+	if runtime.GOOS == "windows" && c.Driver != defaultWindowsDriver {
+		msg := fmt.Sprintf("Plugin only supports the %s driver on Windows, ignoring parameter driver = %s.",
+			defaultWindowsDriver, c.Driver)
+		log.Warning(msg)
+		fmt.Println(msg)
+		c.Driver = defaultWindowsDriver
+	}
+
+	log.WithFields(log.Fields{
+		"driver":    c.Driver,
+		"log_level": *logLevel,
+		"config":    *configFile,
+	}).Info("Starting plugin ")
+
+	if c.Driver == PhotonDriver && err == nil {
```

Comment:
ah, here is another decision. Is there a way to make these checks into a driver-specific method ?